app/controllers/blogs_controller.rb

```diff
@@ -43,11 +49,13 @@ def destroy
 
   private
 
-  def set_blog
-    @blog = Blog.find(params[:id])
+  def set_current_user_blog
+    @blog = current_user.blogs.find(params[:id])
   end
 
   def blog_params
-    params.require(:blog).permit(:title, :content, :secret, :random_eyecatch)
+    ok_items = %i[title content secret]
+    ok_items.push(:random_eyecatch) if current_user.premium
```

Comment:
```suggestion
    ok_items.push(:random_eyecatch) if current_user.premium?
```

booleanのカラムは ? を付けてbooleanであることをわかりやすくすることが多いです

Reply:
?付きに変更しました。